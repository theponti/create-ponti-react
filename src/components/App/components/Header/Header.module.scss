.appName {
    display: flex;
    align-items: center;
    font-size: 24px;
    font-weight: 500;
    color: black;
    text-decoration: none;

    img {
      height: 2.5rem;
      margin-right: 0.4rem;
    }
}

.leftNav,
.rightNav {
    display: flex;
    align-items: center;
}

.menuItem {
  font-weight: 500;
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 70px;
  height: 40px;
  margin: 0 1rem;
  padding: 8px 12px;
  border-radius: 4px;
  color: black;

  &:hover,
  &:focus {
    background-color: rgb(225, 225, 225);
  }
}
