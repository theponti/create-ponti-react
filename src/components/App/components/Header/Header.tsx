import styled from '@emotion/styled';
import PropTypes, { InferProps } from 'prop-types';

import Button from 'components/Button';
import Link from 'components/Link';
import { APP_NAME, ROUTES } from 'services/constants';

import styles from './Header.module.scss';

const Wrap = styled.header`
  display: flex;
  align-content: center;
  justify-content: space-between;
  margin: 0 auto;
  max-width: ${(props) => props.theme.breakpoints.xLarge}px;
  padding: 16px 0;
`;

function Header({ isAuthenticated, onLogin }: InferProps<typeof Header.propTypes>) {
  return (
    <Wrap>
      <div className={styles.rightNav}>
        <Link to="/" className={styles.appName}>
          {APP_NAME}
        </Link>
      </div>
      <div className={styles.leftNav}>
        {isAuthenticated ? (
          <Link className={styles.menuItem} to={ROUTES.ACCOUNT_PATH}>
            Account
          </Link>
        ) : (
          <Button onClick={onLogin} variant="text">
            Get Started
          </Button>
        )}
      </div>
    </Wrap>
  );
}

Header.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  onLogin: PropTypes.func.isRequired,
};

export default Header;
