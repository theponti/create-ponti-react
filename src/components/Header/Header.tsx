import { useCallback } from 'react';
import { NavLink, useNavigate } from 'react-router-dom';
import { APP_NAME } from 'services/constants';
import { supabase } from 'services/supabase';

type HeaderProps = {
  isAuthenticated: boolean,
};

function Header({ isAuthenticated }: HeaderProps) {
  const navigate = useNavigate();
  const onLogoutClick = useCallback(async () => {
    await supabase.auth.signOut();
    navigate('/');
  }, [navigate]);

  return (
    <div className="navbar bg-base-100">
      <div className="flex-1 text-primary">
        <NavLink className="btn btn-ghost normal-case text-xl" to="/">
          {APP_NAME}
        </NavLink>
      </div>
      <div className="flex-none gap-2">
        {!isAuthenticated ? (
          <NavLink data-testid="loginButton" className="btn" to="/signin">
            Get Started
          </NavLink>
        ) : null}
        {isAuthenticated ? (
          <div className="dropdown dropdown-end">
            <button tabIndex={0} className="btn btn-ghost btn-circle avatar" type="button">
              <div className="w-10 rounded-full">
                <img src="https://placeimg.com/80/80/people" alt="avatar" />
              </div>
            </button>
            <ul tabIndex={0} role="menu" className="mt-3 p-2 shadow menu menu-compact dropdown-content bg-base-100 rounded-box w-52">
              <li>
                <NavLink className="justify-between" to="/account">
                  Account
                </NavLink>
              </li>
              <li>
                <div role="button" onClick={onLogoutClick} onKeyDown={onLogoutClick} tabIndex={0}>
                  Logout
                </div>
              </li>
            </ul>
          </div>
        ) : null}
      </div>
    </div>
  );
}

export default Header;
