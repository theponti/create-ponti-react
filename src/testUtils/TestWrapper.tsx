import { ThemeProvider } from '@emotion/react';
import PropTypes, { InferProps } from 'prop-types';
import { BrowserRouter } from 'react-router-dom';

import theme from 'styles/theme';

function TestWrapper({ children }: InferProps<typeof TestWrapper.propTypes>) {
  return (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        {children}
      </ThemeProvider>
    </BrowserRouter>
  );
}

TestWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

export default TestWrapper as React.FC;
