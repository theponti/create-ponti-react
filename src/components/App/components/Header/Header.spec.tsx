/* eslint-disable react/jsx-handler-names */
import { render } from '@testing-library/react';
import TestWrapper from 'testUtils/TestWrapper';
import {
  beforeEach, describe, expect, it, vi,
} from 'vitest';
import Header from './Header';

describe('<Header/>', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('should render unauthenticated header', () => {
    const { container } = render(
      <TestWrapper>
        <Header isAuthenticated={false} onLogin={vi.fn()} />
      </TestWrapper>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render authenticated header', () => {
    const { container } = render(
      <TestWrapper>
        <Header isAuthenticated onLogin={vi.fn()} />
      </TestWrapper>,
    );
    expect(container).toMatchSnapshot();
  });
});
