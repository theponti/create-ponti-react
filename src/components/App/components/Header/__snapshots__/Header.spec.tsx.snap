// Vitest Snapshot v1

exports[`<Header/> > should render authenticated header 1`] = `
<div>
  <header
    class="css-17ouny1"
  >
    <div
      class="_rightNav_9ej0w_15"
    >
      <a
        class="_wrap_1y6kl_1 _appName_9ej0w_1"
        href="/"
      >
        Ponti UI
      </a>
    </div>
    <div
      class="_leftNav_9ej0w_14"
    >
      <a
        class="_wrap_1y6kl_1 _menuItem_9ej0w_20"
        href="/account"
      >
        Account
      </a>
    </div>
  </header>
</div>
`;

exports[`<Header/> > should render unauthenticated header 1`] = `
<div>
  <header
    class="css-17ouny1"
  >
    <div
      class="_rightNav_9ej0w_15"
    >
      <a
        class="_wrap_1y6kl_1 _appName_9ej0w_1"
        href="/"
      >
        Ponti UI
      </a>
    </div>
    <div
      class="_leftNav_9ej0w_14"
    >
      <button
        class=" css-hcwumx"
        data-testid="button"
        type="button"
      >
        Get Started
      </button>
    </div>
  </header>
</div>
`;
